sale_loyalty_order_line_link/models/sale_order.py

```diff
@@ -140,7 +140,11 @@ class SaleOrderLine(models.Model):
     def write(self, vals):
         """When the reward line is update we should refresh the line links as well"""
         res = super().write(vals)
-        if self.filtered(lambda x: x.is_reward_line and x.loyalty_program_id):
+        if self.filtered(
+            lambda x: x.is_reward_line
+            and x.loyalty_program_id
+            and not x.reward_line_ids
```

Comment:
What if the lines change?

Reply:
If the lines change, the applicable promotions are updated, in case there is any applicable promotion a new line is generated that fulfils the conditions of the if and the old one is deleted, in case they do not fulfil the conditions of any promotion all will be deleted. Moreover, if the applicable promotions have not been manually updated when confirming, the order confirmation will do it. Is there any use case that I'm missing and may be giving an error?